rules/aip0158/response_plural_first_field.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0158
+
+import (
+	"fmt"
+
+	"github.com/gertd/go-pluralize"
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var responsePluralFirstField = &lint.MessageRule{
+	Name:   lint.NewRuleName(158, "response-plural-first-field"),
+	OnlyIf: isPaginatedResponseMessage,
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// Rule check: for all messages that end in Response and contain a next_page_token field.
+		// Throw a linter warning if, the first field in the message is not named according to plural(message_name.to_snake().split('_')[1:-1]).
+		if m.FindFieldByName("next_page_token") == nil {
+			return nil
+		}
+		firstField := m.GetFields()[0]
+		pluralize := pluralize.NewClient()
+
+		// Need to convert name to singular first to support none standard case such as persons, cactuses.
```

Comment:
I suggest to create a util function for the conversion and hide the implementation details, like:
```golang
func toPlural(s string) string {
  ...
}

Reply:
I created a separate file for this since we can use it with other rules ex. https://aip.dev/122 > Collection identifiers- Collection identifiers must be plural. 

However, I found out the lib consider `infos` as a plural form of `info`.
https://play.golang.org/p/l4FrJuok5Rh

Reply:
Thanks! Yes, I remember there are couples of places that we can use your function. And it will be easier for us to switch to another library if desired, or make some manual corrections.

Maybe we should move your function into `rules/internal/utils`. What do you think?

Reply:
Done